docs/proposals/device-crd.md

```diff
@@ -0,0 +1,556 @@
+// service exposed by the device.
+type DevicePropertyVisitor struct {
+  // The protocol type to connect to the device and access the property.
+  // Required.
+  Protocol     ProtocolType      `json:"protocol,omitempty"`
+  // A list of visitors describing the properties which can be accessed with the protocol.
+  // Required.
+  Visitors     []PropertyVisitor `json:"visitors,omitempty"`
+}
+
+// ProtocolType describes the protocol type used to communicate with the device instance.
+type ProtocolType string
+
+const (
+	// Bluetooth Low Energy(BLE) Protocol
+	ProtocolTypeBluetoothLE  ProtocolType = "BluetoothLE"
+	// Zigbee Protocol
+	ProtocolTypeZigbee       ProtocolType = "Zigbee"
+	// Modbus RTU Protocol
+	ProtocolTypeModbusRTU    ProtocolType = "ModbusRTU"
+        // Modbus TCP Protocol
+	ProtocolTypeModbusTCP    ProtocolType = "ModbusTCP"
+	// BACnet Protocol
+	ProtocolTypeBACnet       ProtocolType = "BACnet"
+	// OPC UA Protocol
+	ProtocolTypeOPCUA        ProtocolType = "OPCUA"
+)
+
+// PropertyVisitor contains details like which property can be accessed and the
+// access mechanisms are described via key-value pairs in the metadata.
+type PropertyVisitor struct {
+  // The device property visitor name.
+  // Required.
+  Name         string            `json:"name,omitempty"`
+  // The device property name to be accessed. This should refer to one of the
+  // device properties defined in the device model.
+  // Required.
+  PropertyName string            `json:"propertyName,omitempty"`
+  // Additional metadata about the how to access the device property.
+  // Required.
+  Metadata     map[string]string `json:"metadata,omitempty"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DeviceModel is the Schema for the device model API
+// +k8s:openapi-gen=true
+type DeviceModel struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec DeviceModelSpec `json:"spec,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DeviceModelList contains a list of DeviceModel
+type DeviceModelList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []DeviceModel `json:"items"`
+}
+```
+
+### Device model sample
+```yaml
+apiVersion: devices.kubeedge.io/v1alpha1
+kind: DeviceModel
+metadata:
+  labels:
+    description: 'TI Simplelink SensorTag Device Attributes Model'
+    manufacturer: 'Texas Instruments'
+    model: CC2650
+  name: sensor-tag-model
+spec:
+  properties:
+  - name: temperature
+    description: 'Temperature from Temperature Sensor in Degrees Fahrenheit'
+    metadata:
+      dataType: float
+      accessMode: r
+      unit: 'degree fahrenheit'
+  - name: temperature-enable
+    description: 'Enable Temperature Sensor'
+    metadata:
+      dataType: string
+      accessMode: rw
+  propertyVisitors:
```

Comment:
Please give an example on metadata in propertyVisitors.

Reply:
In property visitors , we don't have metadata, we have `visitorConfig` which is there in example.